packages/core/src/interpreter.ts

```diff
@@ -142,9 +142,9 @@ export class Interpreter<
         return clearTimeout(id);
       }
     } as Clock,
-    logger: global.console.log.bind(console),
+    logger: global?.console.log.bind(console),
```

Comment:
Is this change required though? Are all standard~ global objects not available in this environment? Even `self` and `globalThis`?

Reply:
my fault😢 . this change is not required. 
Only `self` and `globalThis` is undefined. `window` is available.

Reply:
Restored that line. :fire:

Reply:
`globalThis` may be `undefined`

<img width="635" alt="image" src="https://user-images.githubusercontent.com/45113594/157364094-e744bd29-7280-4ccf-9950-1ada5b5ac274.png">
1

Reply:
> window is available.

This is somewhat confusing as on the screenshot shared [here](https://github.com/statelyai/xstate/pull/3133#issuecomment-1061755747) it states that `window` shouldn't be available in this environment 🤔

Reply:
replacing `global` with `window` or `globalThis` may be the right way, but why not use `getGlobal()` directly.

<img width="996" alt="image" src="https://user-images.githubusercontent.com/45113594/157390583-1fd82e3c-fe15-4227-9a5b-22fcbb20b340.png">

Reply:
`global` isn't defined on this file, so it refers to `NodeJS.Global` here(that's the reason no error in TS)

Reply:
> replacing global with window or globalThis may be the right way, but why not use getGlobal() directly.

I think you might have misunderstood me - I'm totally on board with using `getGlobal` here. I'm just wondering what global object is gonna be available in your environment.

Reply:
😅  `global` don't have the `console` object
<img width="1115" alt="image" src="https://user-images.githubusercontent.com/45113594/157431821-112d2d80-7afd-4b20-92f4-1b2135e8a9d6.png">

Reply:
That is... some bizarre env there 😅 The especially surprising bit is this:
```
window "[object Object]"
globalThis "[object Window]"
```

I'm still surprised that `window` is available here as that contradicts the documentation - but, of course, I believe the screenshot that you have shared just now. Is `globalThis.console` available there?

Reply:
>  Is `globalThis.console` available there?

Yep.

Reply:
I've tweaked the fix and pushed it out to your branch. Could you check if this variant works for you?

Reply:
Everything works perfectly. Thank you so much :tada: :rocket: